src/test/java/org/elasticsearch/index/query/SpanTermQueryBuilderTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.spans.SpanTermQuery;
+import org.elasticsearch.common.lucene.BytesRefs;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.equalTo;
+
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+
+public class SpanTermQueryBuilderTest extends BaseTermQueryTestCase<SpanTermQueryBuilder> {
+
+    @Override
+    protected SpanTermQueryBuilder createEmptyQueryBuilder() {
+        return new SpanTermQueryBuilder();
+    }
+    
+    @Override
+    protected SpanTermQueryBuilder createQueryBuilder(String fieldName, Object value) {
+        return new SpanTermQueryBuilder(fieldName, value);
+    }
+
```

Comment:
For the TermQuery that kind of error was always related to using "parser.text()" in the fromXContent methods `else` branch, because otherwise int/float values get always read back as Strings. parser.objectText() does some internal checks for numeric tokens and converts correctly.